### ISSUE TITLE
Creating new project fails with "System.Exception: generating help documentation failed"

### ISSUE THREAD

**Issue by essenciary** [fsprojects/ProjectScaffold#210]
Failure info: 
```
Running build failed.
Error:
System.Exception: generating help documentation failed
  at FSI_0001.Build.generateHelp' (Boolean fail, Boolean debug) [0x00000] in <filename unknown>:0
  at FSI_0001.Build+clo@205-18.Invoke (Microsoft.FSharp.Core.Unit _arg1) [0x00000] in <filename unknown>:0
  at Fake.TargetHelper+targetFromTemplate@195[Microsoft.FSharp.Core.Unit].Invoke (Microsoft.FSharp.Core.Unit unitVar0) [0x00000] in <filename unknown>:0
  at Fake.TargetHelper.runSingleTarget (Fake.TargetTemplate`1 target) [0x00000] in <filename unknown>:0

---------------------------------------------------------------------
Build Time Report
---------------------------------------------------------------------
Target         Duration
------         --------
Clean          00:00:00.0017404
AssemblyInfo   00:00:00.0088883
Build          00:00:05.4029755
CopyBinaries   00:00:00.0053061
RunTests       00:00:00.7069119
CleanDocs      00:00:00.0016905
Total:         00:00:14.7962416
Status:        Failure
---------------------------------------------------------------------
  1) System.Exception: generating help documentation failed
  at FSI_0001.Build.generateHelp' (Boolean fail, Boolean debug) [0x00000] in <filename unknown>:0
  at FSI_0001.Build+clo@205-18.Invoke (Microsoft.FSharp.Core.Unit _arg1) [0x00000] in <filename unknown>:0
  at Fake.TargetHelper+targetFromTemplate@195[Microsoft.FSharp.Core.Unit].Invoke (Microsoft.FSharp.Core.Unit unitVar0) [0x00000] in <filename unknown>:0
  at Fake.TargetHelper.runSingleTarget (Fake.TargetTemplate`1 target) [0x00000] in <filename unknown>:0
```

For more info, the output right before the error was: 

```
Generating '/Users/adrian/Dropbox/projects/ProjectScaffold/docs/tools/../content/index.html'
FSharp.Formatting Critical: 0 :
Processing the file 'docpage' failed
Source written to: '/var/folders/kn/tyx40mnn1354rzt8bd7_dcl40000gn/T/tmp6fa7985a.tmp.cs'
Compilation errors:
 - warning: (0, 0) Assuming assembly reference `FSharp.Core, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' matches assembly `FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. You may need to supply runtime policy
 - error: (0, 0) Assembly `FakeLib, Version=4.4.4.0, Culture=neutral, PublicKeyToken=null' references `FSharp.Core, Version=4.4.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version number than imported assembly `FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

System.Exception: Generating HTML failed.
  at FSharp.Literate.RazorRender.handleCompile[String] (System.String source, Microsoft.FSharp.Core.FSharpFunc`2 f) [0x00000] in <filename unknown>:0
  at FSharp.Literate.RazorRender.ProcessFileModel[IDictionary`2] (System.Type modelType, System.Object model, Microsoft.FSharp.Core.FSharpOption`1 properties) [0x00000] in <filename unknown>:0
  at FSharp.Literate.Templating.generateFile (Microsoft.FSharp.Core.FSharpOption`1 references, System.String contentTag, IEnumerable`1 parameters, Microsoft.FSharp.Core.FSharpOption`1 templateOpt, System.String output, IEnumerable`1 layoutRoots) [0x00000] in <filename unknown>:0
  at FSharp.Literate.Templating.processFile (Microsoft.FSharp.Core.FSharpOption`1 references, FSharp.Literate.LiterateDocument doc, System.String output, FSharp.Literate.ProcessingContext ctx) [0x00000] in <filename unknown>:0
  at FSharp.Literate.Literate.ProcessScriptFile (System.String input, Microsoft.FSharp.Core.FSharpOption`1 templateFile, Microsoft.FSharp.Core.FSharpOption`1 output, Microsoft.FSharp.Core.FSharpOption`1 format, Microsoft.FSharp.Core.FSharpOption`1 formatAgent, Microsoft.FSharp.Core.FSharpOption`1 prefix, Microsoft.FSharp.Core.FSharpOption`1 compilerOptions, Microsoft.FSharp.Core.FSharpOption`1 lineNumbers, Microsoft.FSharp.Core.FSharpOption`1 references, Microsoft.FSharp.Core.FSharpOption`1 fsiEvaluator, Microsoft.FSharp.Core.FSharpOption`1 replacements, Microsoft.FSharp.Core.FSharpOption`1 includeSource, Microsoft.FSharp.Core.FSharpOption`1 layoutRoots, Microsoft.FSharp.Core.FSharpOption`1 generateAnchors, Microsoft.FSharp.Core.FSharpOption`1 assemblyReferences, Microsoft.FSharp.Core.FSharpOption`1 customizeDocument) [0x00000] in <filename unknown>:0
  at <StartupCode$FSharp-Literate>.$Main+processScriptFile@184.Invoke (System.String file, System.String output) [0x00000] in <filename unknown>:0
  at Microsoft.FSharp.Core.FSharpFunc`2[System.String,System.String].InvokeFast[Unit] (Microsoft.FSharp.Core.FSharpFunc`2 func, System.String arg1, System.String arg2) [0x00000] in <filename unknown>:0
  at <StartupCode$FSharp-Literate>.$Main.processDirectory@199 (Microsoft.FSharp.Core.FSharpOption`1 format, Boolean processRecursive, Microsoft.FSharp.Core.FSharpFunc`2 processScriptFile, Microsoft.FSharp.Core.FSharpFunc`2 processMarkdown, System.String indir, System.String outdir) [0x00000] in <filename unknown>:0
  at FSharp.Literate.Literate.ProcessDirectory (System.String inputDirectory, Microsoft.FSharp.Core.FSharpOption`1 templateFile, Microsoft.FSharp.Core.FSharpOption`1 outputDirectory, Microsoft.FSharp.Core.FSharpOption`1 format, Microsoft.FSharp.Core.FSharpOption`1 formatAgent, Microsoft.FSharp.Core.FSharpOption`1 prefix, Microsoft.FSharp.Core.FSharpOption`1 compilerOptions, Microsoft.FSharp.Core.FSharpOption`1 lineNumbers, Microsoft.FSharp.Core.FSharpOption`1 references, Microsoft.FSharp.Core.FSharpOption`1 fsiEvaluator, Microsoft.FSharp.Core.FSharpOption`1 replacements, Microsoft.FSharp.Core.FSharpOption`1 includeSource, Microsoft.FSharp.Core.FSharpOption`1 layoutRoots, Microsoft.FSharp.Core.FSharpOption`1 generateAnchors, Microsoft.FSharp.Core.FSharpOption`1 assemblyReferences, Microsoft.FSharp.Core.FSharpOption`1 processRecursive, Microsoft.FSharp.Core.FSharpOption`1 customizeDocument) [0x00000] in <filename unknown>:0
  at FSI_0003.buildDocumentation () [0x00000] in <filename unknown>:0
  at <StartupCode$FSI_0003>.$FSI_0003.main@ () [0x00000] in <filename unknown>:0
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0
Stopped due to error
```

**Comment by erdeszt** [comment]
Having the same issue with the same stack traces on ```Mono JIT compiler version 4.0.1 (tarball Tue May 12 15:39:23 UTC 2015)```

**Comment by jimmydburr** [comment]
Same problem here as the op.  For reference, I've pasted the complete compiler output of the failing target of the initial build. The problem seems to occur when processing the file 'docpage' (shown below). This problem occurs for me on both Mac w/mono and Windows w/VisualStudio.

This results in a project where Fake can only be used to build certain targets. Build targets for 'GenerateDocs' or 'All' can not be used.

Starting Target: GenerateHelp (==> CleanDocs)
/Users/jburrell/programming/f#/mynewfsproj/docs/content/release-notes.md does not exist.
/Users/jburrell/programming/f#/mynewfsproj/docs/content/license.md does not exist.
fsharpi --define:RELEASE --define:HELP generate.fsx
FSharp.Formatting Information: 0 : FSharp.Formatting Logging setup!
Yaaf.FSharp.Scriping Information: 0 : Yaaf.FSharp.Scripting Logging setup!
Copying file: /Users/jburrell/programming/f#/mynewfsproj/docs/output/img/logo-template.pdn
Copying file: /Users/jburrell/programming/f#/mynewfsproj/docs/output/img/logo.png
Copying styles and scripts: /Users/jburrell/programming/f#/mynewfsproj/docs/output/content/style.css
Copying styles and scripts: /Users/jburrell/programming/f#/mynewfsproj/docs/output/content/style_light.css
Copying styles and scripts: /Users/jburrell/programming/f#/mynewfsproj/docs/output/content/tips.js
Copying styles and scripts: /Users/jburrell/programming/f#/mynewfsproj/docs/output/content/img/github-blue.png
Copying styles and scripts: /Users/jburrell/programming/f#/mynewfsproj/docs/output/content/img/github.png
Generating '/Users/jburrell/programming/f#/mynewfsproj/docs/tools/../content/index.html'
FSharp.Formatting Critical: 0 : 
Processing the file 'docpage' failed
Source written to: '/var/folders/d9/b89d0sw51rs2kpcwj09hmcfr0000gp/T/tmp5b6899b4.tmp.cs'
Compilation errors:
 - warning: (0, 0) Assuming assembly reference `FSharp.Core, Version=4.3.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' matches assembly `FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. You may need to supply runtime policy
 - error: (0, 0) Assembly `FakeLib, Version=4.8.0.0, Culture=neutral, PublicKeyToken=null' references `FSharp.Core, Version=4.4.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' which has a higher version number than imported assembly `FSharp.Core, Version=4.3.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'

System.Exception: Generating HTML failed.
   at FSharp.Literate.RazorRender.handleCompile[a] (System.String source, Microsoft.FSharp.Core.FSharpFunc`2 f) in <filename unknown>:line 0
   at FSharp.Literate.RazorRender.ProcessFileModel[b] (System.Type modelType, System.Object model, Microsoft.FSharp.Core.FSharpOption`1 properties) in <filename unknown>:line 0
   at FSharp.Literate.Templating.generateFile (Microsoft.FSharp.Core.FSharpOption`1 references, System.String contentTag, IEnumerable`1 parameters, Microsoft.FSharp.Core.FSharpOption`1 templateOpt, System.String output, IEnumerable`1 layoutRoots) in <filename unknown>:line 0
   at FSharp.Literate.Templating.processFile (Microsoft.FSharp.Core.FSharpOption`1 references, FSharp.Literate.LiterateDocument doc, System.String output, FSharp.Literate.ProcessingContext ctx) in <filename unknown>:line 0
   at FSharp.Literate.Literate.ProcessScriptFile (System.String input, Microsoft.FSharp.Core.FSharpOption`1 templateFile, Microsoft.FSharp.Core.FSharpOption`1 output, Microsoft.FSharp.Core.FSharpOption`1 format, Microsoft.FSharp.Core.FSharpOption`1 formatAgent, Microsoft.FSharp.Core.FSharpOption`1 prefix, Microsoft.FSharp.Core.FSharpOption`1 compilerOptions, Microsoft.FSharp.Core.FSharpOption`1 lineNumbers, Microsoft.FSharp.Core.FSharpOption`1 references, Microsoft.FSharp.Core.FSharpOption`1 fsiEvaluator, Microsoft.FSharp.Core.FSharpOption`1 replacements, Microsoft.FSharp.Core.FSharpOption`1 includeSource, Microsoft.FSharp.Core.FSharpOption`1 layoutRoots, Microsoft.FSharp.Core.FSharpOption`1 generateAnchors, Microsoft.FSharp.Core.FSharpOption`1 assemblyReferences, Microsoft.FSharp.Core.FSharpOption`1 customizeDocument) in <filename unknown>:line 0
   at <StartupCode$FSharp-Literate>.$Main+processScriptFile@184.Invoke (System.String file, System.String output) in <filename unknown>:line 0
   at Microsoft.FSharp.Core.FSharpFunc`2[T,TResult].InvokeFast[V] (Microsoft.FSharp.Core.FSharpFunc`2 func, Microsoft.FSharp.Core.T arg1, Microsoft.FSharp.Core.TResult arg2) in <filename unknown>:line 0
   at <StartupCode$FSharp-Literate>.$Main.processDirectory@199 (Microsoft.FSharp.Core.FSharpOption`1 format, Boolean processRecursive, Microsoft.FSharp.Core.FSharpFunc`2 processScriptFile, Microsoft.FSharp.Core.FSharpFunc`2 processMarkdown, System.String indir, System.String outdir) in <filename unknown>:line 0
   at FSharp.Literate.Literate.ProcessDirectory (System.String inputDirectory, Microsoft.FSharp.Core.FSharpOption`1 templateFile, Microsoft.FSharp.Core.FSharpOption`1 outputDirectory, Microsoft.FSharp.Core.FSharpOption`1 format, Microsoft.FSharp.Core.FSharpOption`1 formatAgent, Microsoft.FSharp.Core.FSharpOption`1 prefix, Microsoft.FSharp.Core.FSharpOption`1 compilerOptions, Microsoft.FSharp.Core.FSharpOption`1 lineNumbers, Microsoft.FSharp.Core.FSharpOption`1 references, Microsoft.FSharp.Core.FSharpOption`1 fsiEvaluator, Microsoft.FSharp.Core.FSharpOption`1 replacements, Microsoft.FSharp.Core.FSharpOption`1 includeSource, Microsoft.FSharp.Core.FSharpOption`1 layoutRoots, Microsoft.FSharp.Core.FSharpOption`1 generateAnchors, Microsoft.FSharp.Core.FSharpOption`1 assemblyReferences, Microsoft.FSharp.Core.FSharpOption`1 processRecursive, Microsoft.FSharp.Core.FSharpOption`1 customizeDocument) in <filename unknown>:line 0
   at FSI_0003.buildDocumentation () in <filename unknown>:line 0
   at <StartupCode$FSI_0003>.$FSI_0003.main@ () in <filename unknown>:line 0
   at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
   at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) in <filename unknown>:line 0
Stopped due to error
Running build failed.

**Comment by cdrnet** [comment]
// cc @tpetricek

**Comment by tpetricek** [comment]
Is this the same issue as https://github.com/tpetricek/FSharp.Formatting/issues/328 ?

**Comment by jimmydburr** [comment]
@tpetricek Possibly. I'm more an F# new-comer but I can follow the line of reasoning on #328. I may be able to discern if my  issue is a dup or not. I'll give it a go and report back. Thx.

**Comment by jimmydburr** [comment]
@tpetricek Yes! This is the exact same issue as: https://github.com/tpetricek/FSharp.Formatting/issues/328 

Hope this helps. Thx.

**Comment by matthid** [comment]
Can you try the patch in #217?

**Comment by jimmydburr** [comment]
Yes, absolutely, will do and report back. Many thanks!

**Comment by jimmydburr** [comment]
Yes! Patch #217 fixed this issue for me on both OS X (10.10) and Windows (10).

Once again, many thanks.
